### PR TITLE
[lang] Remove wildcard imports

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:* 484

*Description of changes:* 
- Replace the wildcard imports with explicit ones in `lang` package to make the code easier to read.
- Also fixes some redundant import namespacing which was causing conflicts as a result
  of wildcard import.
- Removes unused imports

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlurton** [comment]
I'm not opposed to this, however, FYI, we haven't already done this already done this already due to the multiple branches and the havoc it will likely cause with merge conflicts.

**Comment by am357** [comment]
> I'm not opposed to this, however, FYI, we haven't already done this already done this already due to the multiple branches and the havoc it will likely cause with merge conflicts.

That's fair, however in IMHO, the sooner we get to this the less headache we'll have down the track.

**Review by abhikuhikar — DISMISSED** [review]
We have to do this at some point. Better sooner than later. Successful build is all we need. Ship it.

**Review by alancai98 — COMMENTED** [review]
I agree with what's been said already. This is some important tech debt to clean up sooner rather than later. However, we are currently working to merge one branch with a lot of changes into `main`. To prevent further delays due to merge conflicts, I'd prefer to perform this work after the merge.

We have already held off on performing some other tech debt cleaning to post-merge (e.g. 
- #425 Deprecating `ExprNode` with PIG-generated `PartiqlAst` in the evaluator
- #43 Integrating ktlint into Gradle build
).

I haven't looked too closely at the diffs, but it's entirely possible these changes won't result in many conflicts. We could merge this PR and revert it if we feel there are too many conflicts.

**Comment by am357** [comment]
Good call-outs everyone!

@alancai98 to move away from speculation realm by bringing more data: do we know what parts of codebase contribute to the upcoming merge? As this commit includes almost all files under `./lang`, we'll likely have a better understanding of conflicts by knowing the volume of changes incoming; if this is infeasible, I see the _pushing the changes now and revert later (if needed)_  option being viable; considering having a follow-on action item to merge this commit.

**Comment by alancai98** [comment]
> Good call-outs everyone!
> 
> @alancai98 to move away from speculation realm by bringing more data: do we know what parts of codebase contribute to the upcoming merge? As this commit includes almost all files under `./lang`, we'll likely have a better understanding of conflicts by knowing more about the changes incoming; if this is infeasible, I see the _pushing the changes now and revert later (if needed)_ option being viable; with having a follow-on action item to merge this commit right after, in case we revert.

Pretty much all parts of the codebase will be touched. The major updates will affect the evaluator (`lang/eval`) and the parser (`lang/syntax`). From running the `git merge` into `main`, 222 files will be involved (89 new files, 122 modified, some others deleted and renamed) with 13 conflicts. While running the `git merge` into this branch `remove-wildcard-imports` will result in around the same number of files involved but 54 conflicts.

Based on that, imo, we should hold off on removing wildcard imports till after the merge. There may also be other new files coming in to `main` that will need the wildcard imports removed.

**Comment by am357** [comment]
> Based on that, imo, we should hold off on removing wildcard imports till after the merge. There may also be other new files coming in to `main` that will need the wildcard imports removed.

Thanks for the update @alancai98! I'll set this pull request as blocked until we get the merge in place, then.

**Review by dlurton — APPROVED** [review]

File: lang/src/org/partiql/lang/CompilerPipeline.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/CompilerPipeline.kt:50]: Extra '*'

Comment (alancai98) [inline lang/src/org/partiql/lang/CompilerPipeline.kt:50]: (This might get fixed in https://github.com/partiql/partiql-lang-kotlin/issues/43 when we add ktlint)

Not sure why additional whitespace is being added here and on L132 and L295-296

File: lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt:17]: nit: there's an empty comment block that could be deleted here

File: lang/src/org/partiql/lang/ast/StatementToExprNode.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/ast/StatementToExprNode.kt:104]: nit: extra whitespace added here. Also between L123-131, L154-172

File: lang/src/org/partiql/lang/ast/ast.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/ast/ast.kt:151]: nit: extra whitespace added here

File: lang/src/org/partiql/lang/domains/partiql-domains.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/domains/partiql-domains.kt:45]: Hmm, the `partiql-domains.kt` shouldn't be checked into the repo. From the `.gitignore`: https://github.com/partiql/partiql-lang-kotlin/blob/06a2a5df5fe040dae1a3b236ccda954453ce1c41/lang/src/org/partiql/lang/domains/.gitignore#L1-L2

File: lang/src/org/partiql/lang/errors/ErrorAndErrorContexts.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/errors/ErrorAndErrorContexts.kt:105]: nit (formatting that may be fixed by ktlint later): whitespace could be removed
```suggestion
            PropertyType.LONG_CLASS -> longValue()
            PropertyType.STRING_CLASS -> stringValue()
            PropertyType.INTEGER_CLASS -> integerValue()
            PropertyType.TOKEN_CLASS -> tokenTypeValue()
            PropertyType.ION_VALUE_CLASS -> ionValue()
```

File: lang/src/org/partiql/lang/eval/ExprValueType.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/ExprValueType.kt:38]: nit (may be fixed by ktlint in the future): extra whitespace in these core type definitions, L113-116, L139-145, L151-184

File: lang/src/org/partiql/lang/eval/StructExprValue.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/StructExprValue.kt:34]: nit (may be fixed by ktlint in the future): extra whitespace

File: lang/src/org/partiql/lang/eval/builtins/SubstringExprFunction.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/builtins/SubstringExprFunction.kt:28]: `java.lang.Integer.max` is still imported. Does ``java.lang.Integer.min` need removed?

File: lang/src/org/partiql/lang/util/ExprValueFormatter.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/util/ExprValueFormatter.kt:60]: nit: in the past we had aligned the `when` expression branch arrows together, which proved tedious (e.g. if a new clause gets added that's longer, have to adjust every branch). Consider aligning to after the the LHS expression.

File: lang/test/org/partiql/lang/eval/builtins/ConcatEvaluationTest.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/builtins/ConcatEvaluationTest.kt:17]: nit: there are no uses for this value in the file. Could remove.

File: lang/test/org/partiql/lang/eval/io/DelimitedValuesTest.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/io/DelimitedValuesTest.kt:18]: nit: imports are out of order (i.e. should be with the below `org.partiql` imports)

File: lang/src/org/partiql/lang/syntax/SqlParser.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/syntax/SqlParser.kt]: nit (may be fixed by ktlint in the future): extra whitespace

File: lang/src/org/partiql/lang/syntax/SqlParser.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/syntax/SqlParser.kt]: nit: formatting of this line looks off

File: lang/src/org/partiql/lang/syntax/SqlParser.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/syntax/SqlParser.kt]: nit (may be fixed by ktlint in the future): extra whitespace

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt]: Hmm most of the lines changed in this file (other than the imports) add some extra whitespace. I'm concerned this will make it harder to view the git history/blame for these adjusted lines. Could the extra whitespace be reverted from this PR?

**Review by alancai98 — CHANGES_REQUESTED** [review]
Thanks for putting the effort in removing the wildcard imports! There were certainly a lot of files missing our style standards.

I have some concerns about some of the extra whitespace added. I marked most locations as a nit since they didn't span too many lines of code and could be easily fixed when we add ktlint. However, I think the added whitespace in `EvaluatingCompiler` may make it hard to look at the git history/blame for some of the logic.

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (am357) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt]: Will fix.

File: lang/src/org/partiql/lang/CompilerPipeline.kt
Comment (am357) [reply lang/src/org/partiql/lang/CompilerPipeline.kt:50]: Fixed.

Comment (am357) [reply lang/src/org/partiql/lang/CompilerPipeline.kt:50]: Thanks for pointing these out; neither am I. Fixed!

File: lang/src/org/partiql/lang/ast/StatementToExprNode.kt
Comment (am357) [reply lang/src/org/partiql/lang/ast/StatementToExprNode.kt:104]: Fixed

File: lang/src/org/partiql/lang/ast/ast.kt
Comment (am357) [reply lang/src/org/partiql/lang/ast/ast.kt:151]: Fixed.

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (am357) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt]: Looks like the reason for these was that my IDE's continuation indent size has been 8 spaces. I fixed that.

File: lang/src/org/partiql/lang/syntax/SqlParser.kt
Comment (am357) [reply lang/src/org/partiql/lang/syntax/SqlParser.kt]: Fixed.

File: lang/src/org/partiql/lang/syntax/SqlParser.kt
Comment (am357) [reply lang/src/org/partiql/lang/syntax/SqlParser.kt]: Fixed.

File: lang/test/org/partiql/lang/syntax/SqlLexerTest.kt
Comment (am357) [reply lang/test/org/partiql/lang/syntax/SqlLexerTest.kt:50]: Fixed.

File: lang/test/org/partiql/lang/eval/io/DelimitedValuesTest.kt
Comment (am357) [reply lang/test/org/partiql/lang/eval/io/DelimitedValuesTest.kt:18]: Fixed.

File: lang/src/org/partiql/lang/eval/builtins/SubstringExprFunction.kt
Comment (am357) [reply lang/src/org/partiql/lang/eval/builtins/SubstringExprFunction.kt:28]: Fixed.

File: lang/src/org/partiql/lang/eval/ExprValueType.kt
Comment (am357) [reply lang/src/org/partiql/lang/eval/ExprValueType.kt:38]: Fixed.

File: lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt
Comment (am357) [reply lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt:17]: Fixed.

File: lang/test/org/partiql/lang/syntax/SqlLexerTest.kt
Comment (am357) [reply lang/test/org/partiql/lang/syntax/SqlLexerTest.kt:50]: Fixed.

File: lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt
Comment (am357) [reply lang/src/org/partiql/lang/ast/IsImplictJoinMeta.kt:17]: Fixed.

File: lang/src/org/partiql/lang/util/ExprValueFormatter.kt
Comment (am357) [reply lang/src/org/partiql/lang/util/ExprValueFormatter.kt:60]: Fixed.

File: lang/src/org/partiql/lang/eval/StructExprValue.kt
Comment (am357) [reply lang/src/org/partiql/lang/eval/StructExprValue.kt:34]: Fixed.

File: lang/src/org/partiql/lang/domains/partiql-domains.kt
Comment (am357) [reply lang/src/org/partiql/lang/domains/partiql-domains.kt:45]: This is clearly not working.  On the other hand, it looks like we to remove the file completely.

Comment (am357) [reply lang/src/org/partiql/lang/domains/partiql-domains.kt]: Ok [this](https://stackoverflow.com/questions/49520447/gitignore-not-working-for-nested-files) helped, did the following:
1. Remove the file.
2. Commit the change.
3. re-build to ensure file is created.
4. `git status` to ensure file will not get included.

Comment (alancai98) [reply lang/src/org/partiql/lang/domains/partiql-domains.kt]: Thanks for sorting it out! Looks like the merge PR #517 had inadvertently added it.

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (alancai98) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt]: Ok, it's good we found the cause!

**Review by alancai98 — APPROVED** [review]
:shipit:

**Pull request merged**
